test/test_path_sanitization.rb

```diff
@@ -28,4 +28,35 @@ class TestPathSanitization < JekyllUnitTest
     assert_equal source_dir("files", "hi.txt"),
                  Jekyll.sanitized_path(source_dir, "f./../../../../../../files/hi.txt")
   end
+
+  context "on Windows with absolute path" do
+    setup do
+      @base_path = "D:/demo"
+      @file_path = "D:/demo/_site"
+      allow(Dir).to receive(:pwd).and_return("D:/")
+    end
+
+    should "strip just the clean path drive name" do
+      assert_equal "D:/demo/_site",
+                   Jekyll.sanitized_path(@base_path, @file_path)
+    end
+  end
+
+  context "on Windows with file path has matching prefix" do
+    setup do
+      @base_path = "D:/site"
+      @file_path = "D:/sitemap.xml"
+      allow(Dir).to receive(:pwd).and_return("D:/")
+    end
+
+    should "not strip base path" do
+      assert_equal "D:/site/sitemap.xml",
```

Comment:
@ptoomey3, I'm adding this test based on #5276.

The scenario is if we configured source to D:/ and destination to D:/site, the file sitemap.xml will be created in the source path before copying to destination (i.e copy D:/sitemap.xml to D:/site/sitemap.xml). That is where Jekyll build fail in my test bed.